src/cache/GpuResourceCache.js

```diff
@@ -161,7 +161,13 @@ define([
             var entry = (key instanceof ImageSource)
                 ? this.entries.entryForKey(key.key) : this.entries.entryForKey(key);
 
-            return entry ? entry.resource : null;
+            var resource = entry ? entry.resource : null;
+
+            if (resource instanceof Texture) {
```

Comment:
According to benchmarks, checking for the existence of the method is much faster than using instanceof, at least in FF and Chrome. This seems like a critical enough place to consider it. What do you think?